crates/nargo_cli/src/cli/compile_cmd.rs

```diff
@@ -85,7 +85,9 @@ pub(crate) fn run<B: Backend>(
             );
         }
     } else {
-        let program = compile_circuit(backend, &config.program_dir, &args.compile_options)?;
+        let program = compile_circuit(&config.program_dir, &args.compile_options)?;
+        // TODO: optimize circuit before we update common reference string.
+        // Circuit size will be different to the value used here.
```

Comment:
Currently panicking because of this. We can't serialize the general circuit into bberg structures.